datacompy/core.py

```diff
@@ -816,9 +819,9 @@ def columns_equal(
                         col_2 = col_2.str.strip()
 
                 if ignore_case:
-                    if col_1.dtype.kind == "O":
+                    if col_1.dtype == 'object' and col_1.apply(lambda x: isinstance(x, str)).all():
```

Comment:
This seems incorrect. The `dtype` would return an `O` not `object` AFAIK.

Reply:
Hi @fdosani 

The comment is partially correct. `dtype.kind` returns 'O' for object types, while `dtype` itself returns 'object'. Both `col_1.dtype.kind == "O"` and `col_1.dtype == 'object'` are valid ways to check for object dtypes. The proposed change adds an extra check to ensure all elements are strings, which may be useful depending on the specific requirements of the comparison.

![image](https://github.com/user-attachments/assets/37ec347e-ab87-4086-8ea3-623c5c34ed69)

BR,
Team Ana